tests/1596.ispc

```diff
@@ -0,0 +1,37 @@
+static uniform int sz = sizeof(uniform int);
+
+static uniform int a = 7;
+static uniform int *uniform pa = &a;
+static const uniform int *uniform parr[] = {NULL, &a, &a};
+
+static const uniform int b[] = {2, 9, 3, 4, 8, 5, 6, 3};
+static const uniform int *uniform pb[] = {NULL, b + 5, b};
+static const uniform int *uniform pb1[] = {NULL, b, b};
+
+uniform int sVar = sizeof(varying int64);
+
+uniform int ii[2][3][4] = {{{1, 2, 3, 4}, {1, 2, 3, 4}, {1, 2, 3, 4}}, {{1, 2, 3, 4}, {1, 2, 3, 4}, {1, 2, 3, 4}}};
+uniform int *uniform aaa = &ii[1][2][3];
+
+static uniform int sVar_loc_global = sizeof(varying int8);
+
+export uniform int width() { return programCount; }
+
+export void f_v(uniform float RET[]) {
+    static uniform int sVar_loc = sizeof(varying int8);
+    RET[programIndex] = *pa * *parr[2] * *pb[1] * *pb1[2] * *aaa * sz * sVar * sVar_loc * sVar_loc_global;
+}
+
+// Expected Values
+// *pa = 7
+// *parr[2] = 7
+// *pb[1] = 5
+// *pb1[2] = 2
+// *aaa = 4
+// sz = 4
+// sVar = 8 * WIDTH
+// sVar_loc = 1 * WIDTH
+// sVar_loc_global = 1 * WIDTH
+export void result(uniform float RET[]) {
+    RET[programIndex] = 7 * 7 * 5 * 2 * 4 * 4 * (8 * programCount) * (1 * programCount) * (1 * programCount);
```

Comment:
RET has float type. To represent the value precisely it needs to fit mantissa, which is 23 bits.
In case of programCount = 64 (as in avx512skx-i8x64, for example), the result of this expression will require 34 bits (0x3D4000000).
The test will likely pass, as it's going to be rounded in the same way both times, but it's not very reliable way to test for equality. So please use other formula to compute the  "hash".  Either adding values or adding "weighted" values (1*x1 + 2*x2 + 3*x3 +....) is good enough IMHO.